x-pack/legacy/plugins/security/public/lib/role_utils.ts

```diff
@@ -43,3 +44,24 @@ export function isReadOnlyRole(role: Partial<Role>): boolean {
 export function copyRole(role: Role) {
   return cloneDeep(role);
 }
+
+/**
+ * Creates a deep copy of the role suitable for cloning.
+ *
+ * @param role the Role to clone.
+ */
+export function prepareRoleClone(role: Role): Role {
+  const clone = copyRole(role);
+
+  delete clone.metadata;
+  delete clone.transient_metadata;
+
+  clone.name = i18n.translate('xpack.security.roleUtils.defaultCloneRoleName', {
```

Comment:
I'm starting to wonder if we should just delete the name. If we were to clone let's say the "foo_role", create a "copy_of_foo_role" and then go to clone "foo_role" again, we'd end up with the "copy_of_foo_role" again. This would exacerbate https://github.com/elastic/kibana/issues/34712

Reply:
Yeah we can certainly do that. Chances are they'd want to change the role name anyway